src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc

```diff
@@ -161,14 +161,15 @@ std::optional<edm4eic::MutableCluster> CalorimeterClusterRecoCoG::reconstruct(co
   // Used to optionally constrain the cluster eta to those of the contributing hits
   float minHitEta = std::numeric_limits<float>::max();
   float maxHitEta = std::numeric_limits<float>::min();
-  auto time       = pcl.getHits()[0].getTime();
-  auto timeError  = pcl.getHits()[0].getTimeError();
+  auto time      = 0;
+  auto timeError = 0;
   for (unsigned i = 0; i < pcl.getHits().size(); ++i) {
     const auto& hit   = pcl.getHits()[i];
     const auto weight = pcl.getWeights()[i];
     debug("hit energy = {} hit weight: {}", hit.getEnergy(), weight);
     auto energy = hit.getEnergy() * weight;
     totalE += energy;
+    time += (hit.getTime() * energy - time) / totalE;
```

Comment:
`(T*E - T) / E`? Something is wrong with dimensions in this expression?

Reply:
Right! I was adapting a running update formula without weighting.
```suggestion
    time += (hit.getTime() * energy - time * (totalE - energy)) / totalE;
```

I think it would be useful to have a helper class for running updates of mean and stdev when using weighting.

Reply:
Rederived it.
```suggestion
    time += (hit.getTime() - time) * energy / totalE;
```
Meets with limit criteria that the earlier comment violates:
- `energy = 0` should result in no change